vms/platformvm/utxo/camino_handler_test.go

```diff
@@ -106,3 +109,422 @@ func TestVerifySpendUTXOsWithLocked(t *testing.T) {
+			creds:   []verify.Verifiable{},
+			signers: []verify.State{},
+			producedAmounts: map[ids.ID]uint64{
+				h.ctx.AVAXAssetID: 1,
+			},
+			expectedErr: errWrongCredentials,
+		},
+		"Wrong Number of UTXOs": {
+			utxos: []*avax.UTXO{},
+			ins: []*avax.TransferableInput{{
+				Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+				In: &secp256k1fx.TransferInput{
+					Amt: 1,
+				},
+			}},
+			outs: []*avax.TransferableOutput{},
+			creds: []verify.Verifiable{
+				&secp256k1fx.Credential{
+					Sigs: [][crypto.SECP256K1RSigLen]byte{
+						credentials[0].Sigs[0],
+					},
+				},
+			},
+			signers: []verify.State{
+				&secp256k1fx.TransferOutput{
+					Amt: 1,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: []ids.ShortID{
+							addresses[0],
+						}, Threshold: 1,
+					},
+				},
+			},
+			producedAmounts: map[ids.ID]uint64{
+				h.ctx.AVAXAssetID: 1,
+			},
+			expectedErr: ErrWrongUTXONumber,
+		},
+		"Zero Signers, Threshold=1": {
+			utxos: []*avax.UTXO{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			ins: []*avax.TransferableInput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					In: &secp256k1fx.TransferInput{
+						Amt: 1,
+					},
+				},
+			},
+			outs: []*avax.TransferableOutput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			creds: []verify.Verifiable{
+				&secp256k1fx.Credential{},
+			},
+			signers: []verify.State{
+				&secp256k1fx.TransferOutput{
+					Amt: 1,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: []ids.ShortID{
+							addresses[0],
+						}, Threshold: 1,
+					},
+				},
+			},
+			producedAmounts: map[ids.ID]uint64{},
+			expectedErr:     secp256k1fx.ErrTooFewSigners,
+		},
+		"One Signer, Threshold=1": {
+			utxos: []*avax.UTXO{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			ins: []*avax.TransferableInput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					In: &secp256k1fx.TransferInput{
+						Amt: 1,
+						Input: secp256k1fx.Input{
+							SigIndices: []uint32{1},
+						},
+					},
+				},
+			},
+			outs: []*avax.TransferableOutput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			creds: []verify.Verifiable{
+				&secp256k1fx.Credential{
+					Sigs: [][crypto.SECP256K1RSigLen]byte{
+						credentials[1].Sigs[0],
+					},
+				},
+			},
+			signers: []verify.State{
+				&secp256k1fx.TransferOutput{
+					Amt: 1,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: []ids.ShortID{
+							addresses[0],
+							addresses[1],
+						}, Threshold: 1,
+					},
+				},
+			},
+			producedAmounts: map[ids.ID]uint64{},
+			expectedErr:     nil,
+		},
+		"One Signer, Threshold=2": { // changes
```

Comment:
Ouch, found what is wrong here. It contains misleading sigIndices of two elements.
To make it more to the point, please add additional signature in the credentials (so it corresponds to the sigIndices).

Also thinking more descriptive name of this important test would be
```suggestion
		"More signatures than is enough cause the error": {
```